bundle/src/main/java/com/adobe/acs/commons/ondeploy/scripts/OnDeployScript.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2018 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.ondeploy.scripts;
+
+import com.day.cq.search.QueryBuilder;
+import org.apache.sling.api.resource.ResourceResolver;
+
+/**
+ * An on-deploy script that can be run via the OnDeployExecutor.
+ */
+public interface OnDeployScript {
+    /**
+     * Execute the script, passing in a resourceResolver and queryBuilder instance.
+     *
+     * @param resourceResolver Resource resolver.
+     * @param queryBuilder Query builder.
+     */
+    void execute(ResourceResolver resourceResolver, QueryBuilder queryBuilder);
```

Comment:
Is there a reason why QueryBuilder is explicitly passed in? You could get in the Base via `resourceResolver.adaptTo(QueryBuilder.class)` . Passing this is feels a big awkward IMO since not all scripts will query.

Reply:
how about "because we always did it that way"?

Seriously, this is likely just a vestige of my youth when I didnt realize a `ResourceResolver` could be adapted to a `QueryBuilder`.  Good call on that, I'll add it to my cleanups.

Reply:
ahh - to be young again :)